packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/environmentVariableWalkthrough.ts

```diff
@@ -0,0 +1,161 @@
+          (acc, cur) => ({
+            ...acc,
+            [cur]: { Ref: _.camelCase(cur) },
+          }),
+          {},
+        ),
+      });
+  }
+};
+
+const selectEnvironmentVariableQuestion = async (environmentVariables: object): Promise<'add' | 'update' | 'remove' | 'abort'> => {
+  const { operation } = await inquirer.prompt([
+    {
+      name: 'operation',
+      message: 'Select what you want to do with environment variables:',
+      type: 'list',
+      choices: [
+        {
+          value: 'add',
+          name: 'Add new environment variable',
+        },
+        {
+          value: 'update',
+          name: 'Update existing environment variables',
+          disabled: _.size(environmentVariables) === 0,
+        },
+        {
+          value: 'remove',
+          name: 'Remove existing environment variables',
+          disabled: _.size(environmentVariables) === 0,
+        },
+        {
+          value: 'abort',
+          name: "I'm done",
+        },
+      ],
+    },
+  ]);
+
+  return operation;
+};
+
+const addEnvironmentVariableQuestion = async (
+  environmentVariables: object,
+): Promise<{ newEnvironmentVariableKey: string; newEnvironmentVariableValue: string }> => {
+  const { newEnvironmentVariableKey, newEnvironmentVariableValue } = await inquirer.prompt([
+    {
+      name: 'newEnvironmentVariableKey',
+      message: 'Enter the environment variable name:',
+      type: 'input',
+      validate: input => {
+        if (input.match(/^[a-zA-Z0-9_]+$/) === null) {
+          return 'Use only /^[a-zA-Z0-9_]+$/';
+        }
+        if (Object.keys(environmentVariables).includes(input)) {
+          return `Key "${input}" is used already`;
+        }
+        return true;
+      },
+    },
+    {
+      name: 'newEnvironmentVariableValue',
+      message: 'Enter the environment variable value:',
+      type: 'input',
+      validate: input => input.length <= 2048,
+    },
+  ]);
+  return {
+    newEnvironmentVariableKey,
+    newEnvironmentVariableValue,
+  };
+};
+
+const updateEnvironmentVariableQuestion = async (
+  environmentVariables: object,
+): Promise<{ newEnvironmentVariableKey: string; newEnvironmentVariableValue: string; targetedKey: string }> => {
+  const { targetedKey } = await inquirer.prompt([
+    {
+      name: 'targetedKey',
+      message: 'Which environment variable do you want to update:',
+      type: 'list',
+      choices: Object.keys(environmentVariables),
+    },
+  ]);
+
+  const { newEnvironmentVariableKey, newEnvironmentVariableValue } = await inquirer.prompt([
+    {
+      name: 'newEnvironmentVariableKey',
+      message: 'Enter the environment variable name:',
+      type: 'input',
+      validate: input => {
+        if (input.match(/^[a-zA-Z0-9_]+$/) === null) {
+          return 'Use only /^[a-zA-Z0-9_]+$/';
+        }
+        if (Object.keys(environmentVariables).includes(input) && input !== targetedKey) {
```

Comment:
Same comment here about using `hasOwnProperty()` here.

Reply:
Right! Let me use `_.has` which is used in the other palces.